cmd/clusterctl/pkg/client/repository/repository_local_unix.go

```diff
@@ -0,0 +1,215 @@
+
+	var err error
+
+	u, err := url.Parse(providerConfig.URL())
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid url")
+	}
+	absPath := u.Path
+	if u.RawPath != "" {
+		absPath = u.RawPath
+	}
+
+	if !filepath.IsAbs(absPath) {
+		return nil, errors.Errorf("invalid path: path %q must be an absolute path", providerConfig.URL())
+	}
+
+	parts := strings.Split(providerConfig.URL(), "/")
+	// find the first occurance of provider name and use it as a seperator between base and layout
+	providerNameIndex := 0
+	for i, p := range parts {
+		if p == providerConfig.Name() {
+			providerNameIndex = i
+		}
+	}
+	if providerNameIndex < 1 {
+		return nil, errors.Errorf("invalid path: path %q must contain provider name %q", providerConfig.URL(), providerConfig.Name())
+	}
+
+	basePath := filepath.Join(parts[:providerNameIndex-1]...)
+	basePath = filepath.Clean("/" + basePath) // ensure basePath starts with "/"
+	layoutPath := filepath.Join(parts[providerNameIndex-1:]...)
+	layoutParts := strings.Split(layoutPath, "/")
+	// Check if the layoutPath is in the expected format.
+	if len(layoutParts) < 5 {
+		return nil, errors.Errorf("invalid path: path should be in the form {basepath}/{owner}/{provider-name}/releases/{version}/{path/components.yaml}")
+	}
+	owner := layoutParts[0]
+	providerName := layoutParts[1]
+	defaultVersion := layoutParts[3]
+	path := filepath.Join(layoutParts[4:]...)
+	// rootpath is the relative path starting after {version} directory and excludes component fiename
+	rootPath, componentsPath := filepath.Split(path)
+
+	repo := &localRepository{
+		providerConfig:        providerConfig,
+		configVariablesClient: configVariablesClient,
+		basepath:              basePath,
+		owner:                 owner,
+		providerName:          providerName,
+		defaultVersion:        defaultVersion,
+		rootPath:              rootPath,
+		componentsPath:        componentsPath,
+	}
+
+	if defaultVersion == "latest" {
+		repo.defaultVersion, err = repo.getLatestRelease()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get latest version")
+		}
+	}
+
+	return repo, nil
+}
+
+// getLatestRelease returns the latest release for the local repository.
+func (r *localRepository) getLatestRelease() (string, error) {
+
+	// get all the sub-directories under {releases} directory
+	releasesPath := filepath.Join(r.basepath, r.owner, r.providerName, "releases")
+	files, err := ioutil.ReadDir(releasesPath)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to list release directories")
+	}
+	var releases []string
+	for _, f := range files {
+		if f.IsDir() {
+			releases = append(releases, f.Name())
+		}
+	}
+
+	// search for the latest release according to semantic version
+	// releases with names that are not semantic version number are ignored
+	var latestTag string
+	var latestReleaseVersion *version.Version
+	for _, r := range releases {
```

Comment:
What about merging this for loop with the one above?  If I'm not wrong, you need to add a test for ignoring what is not a dir before the  "is not a semver" test...

Reply:
Done